addOns/pscanrulesBeta/src/main/java/org/zaproxy/zap/extension/pscanrulesBeta/SiteIsolationScanRule.java

```diff
@@ -83,17 +83,13 @@ public class SiteIsolationScanRule extends PluginPassiveScanner
 
     @Override
     public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
-
-        // Specs don't state that errors pages should be excluded
-        // However, successful responses are associated to a resource
-        // that should be protected.
-        // Only consider HTTP Status code 2XX to avoid a False Positive
-        if (!HttpStatusCode.isSuccess(msg.getResponseHeader().getStatusCode())
+        // Specs don't state that errors pages should be excluded. However, successful responses are
+        // associated to a resource that should be protected, while error pages are not. Therefore,
+        // only consider HTTP Status code 2XX to avoid a False Positive
+        if (HttpStatusCode.isSuccess(msg.getResponseHeader().getStatusCode())
                 || getHelper().isPage200(msg)) {
```

Comment:
The correct is to call `!isSuccess(msg)`.

Reply:
So, I changed the flow from `if !A return else do_something()` to `if A do_something()`. This because `A` here was what is causing the bug; namely that `if isPage200(...)` the `HttpMessage` is skipped. So I believe this should either be:
```
if (!HttpStatusCode.isSuccess(msg.getResponseHeader().getStatusCode())
        && !getHelper().isPage200(msg)) {
        return;
}

rules.forEach(...);
```
or (what I wrote)
```
if (HttpStatusCode.isSuccess(msg.getResponseHeader().getStatusCode())
        || getHelper().isPage200(msg)) {
        rules.forEach(...);
}
```
That is; only scan pages that have status code 200-299 or are configured as non-standard error handling conditions as "page200" (see https://www.zaproxy.org/docs/desktop/start/features/custompages/). We could discuss whether this is what we want it to be, but at least we won't have so many false negatives as we have now.

Reply:
Might be that the UI is not that clear but I highlighted the original code, which would be lines 91 to 90. I'm saying that the original code should have been (and should be) `!isSuccess(msg)`. Note that `isSuccess(HttpMessage)` is other method than `HttpStatusCode.isSuccess(HttpMessage)`.

Reply:
Ah, thanks that makes sense. Shall I change my PR up s.t. it uses `isSuccess(msg)`? I think that this is exactly what I was aiming for all along

Reply:
Changed it! Also fixed the tests by (re)adding mocks/stubs for `isSuccess` as it didn't seem to work otherwise